src/sacn/source.c

```diff
@@ -595,13 +1232,695 @@ etcpal_error_t sacn_source_reset_networking(sacn_source_t handle, uint16_t unive
+
+        if (netint_state->num_refs == 0)
+          remove_sacn_source_netint(source, &netint_state);
+      }
+    }
+
+    remove_sacn_source_universe(source, &universe, universe_iter);
+    universe_incremented = true;
+  }
+
+  return universe_incremented;
+}
+
+// Needs lock
+void process_universe_null_pap_transmission(SacnSource* source, SacnSourceUniverse* universe)
+{
+  // If 0x00 data is ready to send
+  if (universe->has_null_data && ((universe->null_packets_sent_before_suppression < NUM_PRE_SUPPRESSION_PACKETS) ||
+                                  etcpal_timer_is_expired(&universe->null_keep_alive_timer)))
+  {
+    // Send 0x00 data & reset the keep-alive timer
+    send_null_data_multicast(source, universe);
+    send_null_data_unicast(source, universe);
+    process_null_sent(universe);
+    etcpal_timer_reset(&universe->null_keep_alive_timer);
+  }
+#if SACN_ETC_PRIORITY_EXTENSION
+  // If 0xDD data is ready to send
+  if (universe->has_pap_data && ((universe->pap_packets_sent_before_suppression < NUM_PRE_SUPPRESSION_PACKETS) ||
+                                 etcpal_timer_is_expired(&universe->pap_keep_alive_timer)))
+  {
+    // Send 0xDD data & reset the keep-alive timer
+    send_pap_data_multicast(source, universe);
+    send_pap_data_unicast(source, universe);
+    process_pap_sent(universe);
+    etcpal_timer_reset(&universe->pap_keep_alive_timer);
+  }
+#endif
+}
+
+// Needs lock
+void send_data_multicast_ipv4_ipv6(const SacnSource* source, SacnSourceUniverse* universe, const uint8_t* send_buf)
+{
+  // Send multicast on IPv4 and/or IPv6
+  if (!universe->send_unicast_only)
+  {
+    if ((source->ip_supported == kSacnIpV4Only) || (source->ip_supported == kSacnIpV4AndIpV6))
+      sacn_send_multicast(universe->universe_id, kEtcPalIpTypeV4, send_buf, universe->netints, universe->num_netints);
+    if ((source->ip_supported == kSacnIpV6Only) || (source->ip_supported == kSacnIpV4AndIpV6))
+      sacn_send_multicast(universe->universe_id, kEtcPalIpTypeV6, send_buf, universe->netints, universe->num_netints);
+  }
+}
+
+// Needs lock
+void send_data_unicast_ipv4_ipv6(const SacnSource* source, SacnSourceUniverse* universe, const uint8_t* send_buf)
+{
+  // Send unicast on IPv4 and/or IPv6
+  if ((source->ip_supported == kSacnIpV4Only) || (source->ip_supported == kSacnIpV4AndIpV6))
+    send_data_unicast(kEtcPalIpTypeV4, send_buf, &universe->unicast_dests);
+  if ((source->ip_supported == kSacnIpV6Only) || (source->ip_supported == kSacnIpV4AndIpV6))
+    send_data_unicast(kEtcPalIpTypeV6, send_buf, &universe->unicast_dests);
+}
+
+// Needs lock
+void increment_sequence_number(SacnSourceUniverse* universe)
+{
+  ++universe->seq_num;
+  universe->null_send_buf[SACN_SEQ_OFFSET] = universe->seq_num;
+#if SACN_ETC_PRIORITY_EXTENSION
+  universe->pap_send_buf[SACN_SEQ_OFFSET] = universe->seq_num;
+#endif
+}
+
+// Needs lock
+void send_null_data_multicast(const SacnSource* source, SacnSourceUniverse* universe)
```

Comment:
Are these functions really necessary if their only job is to call another function?

Reply:
Not really, I'll go ahead and remove them.

Reply:
Resolved in https://github.com/ETCLabs/sACN/pull/32/commits/38cae99295d745bd5892951c99ccd3e0d377bef7.